main/streams/userspace.c

```diff
@@ -510,12 +513,23 @@ PHP_FUNCTION(stream_wrapper_unregister)
 		RETURN_THROWS();
 	}
 
+	php_stream_wrapper *wrapper = zend_hash_find_ptr(php_stream_get_url_stream_wrappers_hash(), protocol);
 	if (php_unregister_url_stream_wrapper_volatile(protocol) == FAILURE) {
 		/* We failed */
 		php_error_docref(NULL, E_WARNING, "Unable to unregister protocol %s://", ZSTR_VAL(protocol));
 		RETURN_FALSE;
 	}
 
+	ZEND_ASSERT(wrapper != NULL);
+	if (wrapper->wops == &user_stream_wops) {
+		struct php_user_stream_wrapper *uwrap = (struct php_user_stream_wrapper *)wrapper;
+		zend_resource *resource = uwrap->resource;
+		if (GC_DELREF(resource) == 0) {
+			// uwrap will be released by resource destructor
+			rc_dtor_func((zend_refcounted *)resource);
+		}
+	}
```

Comment:
Can't you use something more standard like zend_list_delete here?

Reply:
Ah, I didn't realize `EG(regular_list)` had a destructor attached to it that will free the resource. Thanks!

Reply:
phpitnernalsbook.com has a good [chapter about resources](https://www.phpinternalsbook.com/php7/internal_types/zend_resources.html).